src/core/service.c

```diff
@@ -163,12 +161,10 @@ static int service_set_main_pid(Service *s, pid_t pid) {
 
         s->main_pid = pid;
         s->main_pid_known = true;
+        s->main_pid_alien = pid_is_my_child(pid) == 0;
```

Comment:
I think we should probably change this from a `==` check to a `<=` check... if we can't determine this we should assume the process is alien I think

Reply:
If `pid_is_my_child()` fails then it's mostly probably due to the fact that pid has exited and has been reaped so that would indeed means the process was alien.

OTOH if we know that the main pid has exited we should stop the unit it belongs to...

This change would need a separate commit anyway since that would be a behavior change.